app/gui/src/controller/graph/widget.rs

```diff
@@ -392,7 +392,7 @@ impl QueryData {
     /// Generate visualization metadata for this query.
     fn visualization_metadata(&self) -> Metadata {
         let arguments: Vec<Code> = vec![
-            Self::escape_visualization_argument(&self.method_name).into(),
+            format!("(self.{}...)", &self.method_name).into(),
```

Comment:
It is surprising to me that this could work. I assumed that the arguments passed to the visualization preprocessor were treated as expressions in method call argument position. In that case, `self` would not refer to the right thing. Looks like my assumption was incorrect.

So, to be clear, is it true that the passed-in "arguments" are treated effectively as expressions within the BODY of the visualization preprocessor method?

Reply:
While it will surely work when sketching out that query flow in the IDE itself, the question is if it will be resolved as intended through the visualization API. You would need to check the log of visualization requests and responses to be sure it behaves as expected.

The IDE is able to come up with a decent set of widgets purely based on data from suggestion database that it already has, even when the visualization query fails. So inspecting the graph visually for dropdowns is not enough to assess that it works.

Reply:
How does the `self` value get in, @4e6? There is:
```scala
         ctx.executionService.callFunctionWithInstrument(
            visualisation.cache,
            visualisation.module,
            visualisation.callback,
            expressionValue +: visualisation.arguments: _*
          )
```
in `ProgramExecutionSupport.scala` and `expressionValue` is `self` while the rest is defined by the visualization.

Reply:
> Is there a way to obtain the name of the self node? In this case operator18?

Would referencing it even work? I'm under the impression that the visualization would not see the main method's inner scope. Either way, the `self` argument can be an arbitrary expression, It doesn't have to be a single variable. That is the case for example in chained applications (`"-a~".trim what="-" . trim what="~"`).

> Can we at least find the type of operator18 out

To some degree, yes. As the IDE receives every evaluated expression's type in an `expressionUpdate` message. But this seems redundant, I'm already sending the relevant expression ID to the language server. I shouldn't need to feed it back with the information i received from it in the first place.

Reply:
To continue experiments I'd like to change the IDE to send invoke the `WIDGET_VISUALIZATION_METHOD` with one more argument - reference (FQN name is sufficient) to the module that is being edited. Help needed - nothing in `widget.rs` file seems to have a reference to the module name as far as I can tell.

Reply:
However IDE cannot specify the name of the module that requests the visualization, as the IDE doesn't know it. According to @Frizi, following might help:

> Interesting change: https://github.com/enso-org/enso/pull/7028/files#diff-4b79b29711f3cafb4bc4dbbd0882b63066d1de06e338bc8f86bf7c03a3e2e936R45
The project name will now actually be available in the graph.

Reply:
@Frizi - that commit helped a lot. I am able to send the name of the edited module to the engine and evaluate the visualization properly in the context of such `visualisation_module`. Please help me with reviews and QA in #7115